### PR TITLE
Switch to last used tab

### PR DESCRIPTION
**Pull request by pikou1995**
#320 😁

### CODE REVIEW

File: app/manifest.json [hunk 1]
```diff
@@ -74,7 +74,8 @@
     "49-showlatestdownload": {"description": "DOWNLOADS: Show latest download"},
     "50-capturescreenshot": {"description": "UTILITY: Capture current viewport screenshot"},
     "51-capturefullsizescreenshot": {"description": "UTILITY: Capture full size screenshot (max height is 16,348px due to browser limitation)"},
-    "52-forcecapturefullsizescreenshot": {"description": "UTILITY: Force capture full size screenshot (when window is not scrollable)"}
+    "52-forcecapturefullsizescreenshot": {"description": "UTILITY: Force capture full size screenshot (when window is not scrollable)"},
+    "53-lastusedtab": {"description": "TABS: Switch to last used tab"}
```
Comment (mikecrittenden) [inline app/manifest.json:78]: It's been a while, but IIRC these are sorted alphabetically in the settings screen, meaning if you change this to `05-lastusedtab` then it'll show up right after the ones for next and previous tab which is closer to where users would expect to find it.

File: app/scripts/actions/lastUsedTab.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * This queue stores two tabs - last tab and current tab.
+ * @type {chrome.tabs.TabActiveInfo[]}
+ */
+let usedTabInfoQueue = [undefined, undefined]
+
+/**
+ * Manage usedTabInfoQueue.
+ */
+browser.tabs.onActivated.addListener(function (currentTab) {
+  usedTabInfoQueue = [usedTabInfoQueue[1], currentTab]
```
Comment (mikecrittenden) [inline app/scripts/actions/lastUsedTab.js:11]: Nice! Very tricky.

File: app/scripts/background.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 'use strict'
 
 import captureScreenshot from './actions/captureScreenshot'
+import lastUsedTab from './actions/lastUsedTab'
```
Comment (mikecrittenden) [inline app/scripts/background.js:4]: Great refactor, having these separate files for actions.

**Review by pikou1995 — COMMENTED** [review]
~~Wait a moment, it miss one situation.~~ done.

File: app/scripts/actions/lastUsedTab.js (hunk 2 above)
Comment (pikou1995) [reply app/scripts/actions/lastUsedTab.js:12]: Last commit takes switch window into consideration.

File: app/manifest.json (hunk 1 above)
Comment (pikou1995) [reply app/manifest.json:78]: Will it make a user's config loss if I sort command list? in `chrome://extensions/shortcuts`.

Comment (pikou1995) [reply app/manifest.json]: I've inserted this command after `14-lasttab`

**Comment by mikecrittenden** [comment]
I love it! Thank you so much!

**Pull request merged**